Lib/fractions.py

```diff
@@ -567,11 +567,15 @@ def __hash__(self):
         # dinv is the inverse of self._denominator modulo the prime
         # _PyHASH_MODULUS, or 0 if self._denominator is divisible by
         # _PyHASH_MODULUS.
-        dinv = pow(self._denominator, _PyHASH_MODULUS - 2, _PyHASH_MODULUS)
+        try:
+            dinv =  pow(self._denominator, -1, _PyHASH_MODULUS)
+        except ValueError:
```

Comment:
In what case `ValueError` is raised?

Reply:
`ValueError` is raised if and only if `_PyHASH_MODULUS` divides `self._denominator`.  There is no modular inverse if and only if the gcd of the base and the modulus is non-trivial (not 1), but the modulus is a prime in this case, so in this case that's the same as the base being a multiple of the modulus.

Reply:
Thank you. We need a test for this.